### PR TITLE
feat(toc): add toc to api page

### PR DESCRIPTION
**Pull request by amcdnl**
This PR:
- Adds Table of Contents to the API page
- Cleans up table styles on dark themes
- Cleans up CSS positioning on headers
- Adds border accents to h3s in the API page
- Misc font / line height normalization
- Increases TOC width size
- Adds ellipsis to TOC content if one word overflowing
- Removes unneeded element creation during content bootstrap

It requires https://github.com/angular/material2/pull/7827 in order to function.

img

### CODE REVIEW

File: src/app/pages/component-viewer/_component-viewer-theme.scss [hunk 1]
```diff
@@ -14,6 +14,9 @@
     .mat-tab-label:focus {
       color: mat-color($foreground, text);
     }
+  }
 
+  .docs-api-h3 {
+    border-bottom: 1px solid rgba(mat-color($foreground, secondary-text), 0.15);
```
Comment (willshowell) [inline src/app/pages/component-viewer/_component-viewer-theme.scss]: How about?

```scss
1px solid mat-color($foreground, divider);
```

Comment (jelbourn) [reply src/app/pages/component-viewer/_component-viewer-theme.scss]: Yeah, that's a little more in line with intention

File: src/styles/_markdown.scss [hunk 2]
```diff
@@ -64,23 +64,18 @@ pre {
...
-  // deduct -30px so the anchor icon will be positioned outside
-  // the content and pad it 30px so the header will set inline
-  // with the content displayed, absolute position will make
-  // it where you can't hover the header
-  padding-left: 30px;
-  margin-left: -30px;
-  position: relative;
+  a {
+    text-decoration: none;
+    margin-left: -30px;
```
Comment (jelbourn) [inline src/styles/_markdown.scss]: Keep the comment?

**Comment by amcdnl** [comment]
Ready for another review.

**Comment by jelbourn** [comment]
@amcdnl just needs rebase

**Pull request merged**